### ISSUE TITLE
Incorrect error in declaration emit for set accessors

### ISSUE THREAD

**Issue by JsonFreeman** [Microsoft/TypeScript#1976]
Compile the following with -declaration:

``` ts
module M {
    interface I { }
    export class C {
        set foo(x: I) { }
    }
}
```

The error says, "Parameter 'foo' of public property setter from exported class has or is using private name 'I'."

But it should not be parameter 'foo', it should be parameter 'x'.

**Comment by NaridaL** [comment]
For comparison, the error if the return type of a getter is inaccessible doesn't mention anything except the type itself:

`Return type of public static property getter from exported class has or is using private name '{0}'.`

I'd suggest bringing the setter error in line with that, which prop is the problem is obvious from the location of the error, and it's a setter -- which parameter it is isn't a big mystery.

> Parameter of public property setter from exported class has or is using private name 'I'.

**Comment by mhegazy** [comment]
sounds like a good proposal.
